docs/release_process.rst

```diff
@@ -15,79 +14,112 @@ number occurs.
+4. Get a positive review.
 
-5. `Create new version release to PYPI`_.
+5. Merge version bump branch into the ``release`` branch.
 
-Major and Minor Releases
-^^^^^^^^^^^^^^^^^^^^^^^^
-Major and minor releases may add one or more trivial or non-trivial features
-and functionalities.
+6. Tag the resulting merge commit.
+   See `Tag new version`_ below for how to do this.
 
-1. Create a new branch off of the ``master`` named something like
-   ``release-[major,minor]-{NEW_VERSION}``.
+7. As a repository administrator, merge the ``release`` branch into ``master``
+   locally and push the updated ``master`` to upstream. (Replace "upstream"
+   in the example below with your applicable remote name.)
+
+   .. code-block:: bash
+
+      $ git fetch --all
+      $ git checkout upstream/master
+      $ git merge --log --no-ff upstream/release
+      $ git push upstream master
+
+7. Update version reference in the `XAITK/xaitk.github.io home page`_ to the
+   new version.
+
+Patch Release
+^^^^^^^^^^^^^
+A patch release should only contain fixes for bugs or issues with an existing
+release.
+No new features or functionality should be introduced in a patch release.
+As such, patch releases should only ever be based on an existing release point
+(git tag).
+
+This list assumes we are creating a new patch release off of the *latest*
+release version, i.e. off of the ``release`` branch.
+If a patch release for an older release version is being created, see the
+`Patching an Older Release`_ section.
+
+1. Create a new branch off of the ``release`` branch named something like
+   ``update-to-v{NEW_VERSION}``, where ``NEW_VERSION`` is the target ``X.Y.Z``,
+   including the bump in the patch (``Z``) version component.
 
-   a. Increment patch value in  ``pyproject.toml`` file's ``version`` attribute
-      under the `[tool.poetry]` section.
+   a. Use the ``scripts/update_release_notes.sh`` script to update the project
+      version number, create ``docs/release_notes/v{NEW_VERSION}.rst``, and add
+      a new pending release notes stub file. E.g.
 
-      * See `Poetry's version command`_ for a convenient means of incrementing
-        the version.
+      .. code-block:: bash
 
-   b. Rename the ``docs/release_notes/pending_release.rst`` file to
-      ``docs/release_notes/v{VERSION}.rst``, matching the new version value.
-      Add a descriptive paragraph under the title section summarizing this
-      release.
+         $ ./scripts/update_release_notes.sh patch
 
-   c. Add a reference to the new release notes RST file in
-      ``docs/release_notes.rst``.
+   b. Add a descriptive paragraph under the title section of
+      ``docs/release_notes/v{NEW_VERSION}.rst`` summarizing this release.
 
-   d. In a separate commit, add back a blank pending release notes file stub.
-      See `Stub Pending Notes File`_.
+2. Push the created branch to the upstream repository, not your fork (this is
+   an exception to the normal forky workflow).
```

Comment:
Pending change above, should also update here:
```suggestion
   an exception to the normal fork workflow).
```

Reply:
Done.